### PR TITLE
DFR 3153 Contested Form A Smoke test

### PR DESCRIPTION
**Pull request by al-hmcts**
### Jira link

[DFR-3153](https://tools.hmcts.net/jira/browse/DFR-3153)

### Change description

Add Contested From A case submission smoke Test

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

**Comment by al-hmcts** [comment]
**Note:** Not 100% set on the most appropriate way to retrieve view components (byRole, Id, Name, Group etc..) but I think by ID is the way to go (and least likely to change and make the tests flakey), however there are some components without IDs or where pulling the ID does not provide us with a usable component. 

As such there are a few inconsistencies in pulling view components between pages where we tried different approaches, but for the sake of getting the PR in I thought I will open it up to the group to get thoughts and then refactor appropriately when I come to do the consented journey (which shares pages with the contested).

File: playwright-e2e/pages/create-case/FinancialRemedyCourtPage.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        this.frcReasonTxtBox = page.locator('#reasonForFRCLocation');
+        this.reasonForHomeCourtTxtBox = page.locator('#reasonForLocalCourt');
+    }
+
+    async selectCourtZoneDropDown() {
+        await this.courtZoneDropDown.selectOption(this.courtRegion);
+
+        this.frcDropDown = this.page.locator(`#${this.courtRegion.toLowerCase()}FRCList`);
+        await expect(this.frcDropDown).toBeVisible();
+        await this.frcDropDown.selectOption(`${this.courtFrc} FRC`);
```
Comment (so99y) [inline playwright-e2e/pages/create-case/FinancialRemedyCourtPage.ts:39]: why don't move " FRC" to the value of `this.courtFrc` as well?

Comment (so99y) [inline playwright-e2e/pages/create-case/FinancialRemedyCourtPage.ts:39]: worth adding a new constants for "Nottingham FRC" due to flexibility.

File: playwright-e2e/pages/create-case/FinancialAssetsPage.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    async insertFamilyHomeValue(value: string) {
+        await this.netFamilyHomeValueTxtBox.fill(value);
+    }
+
+    async checkPotentialIssueNotApplicableCheckbox(){
+        await this.potentialIssuesNotApplicableCheckbox.check();
+    }
+
+
```
Comment (so99y) [inline playwright-e2e/pages/create-case/FinancialAssetsPage.ts:37]: remove extra blank lines.

File: playwright-e2e/pages/helpers/SolicitorDetailsHelper.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { Locator, Page } from "playwright";
+import { expect } from "playwright/test";
+
+export class SolicitorDetailsHelper {
+
+    async selectOrgansiation(page: Page, orgName: string) {
```
Comment (so99y) [inline playwright-e2e/pages/helpers/SolicitorDetailsHelper.ts:6]: typo

File: test/helpers/PlaywrightHelper.js [hunk 4]
```diff
@@ -58,7 +58,7 @@ class PlaywrightHelper extends Helper {
       if (Array.isArray(locator)) {
         for (let i = 0; i < locator.length; i++) {
           // eslint-disable-next-line no-await-in-loop
-          await page.waitForSelector(this.getEnabledCssLocator(locator[i]), { visible: true, timeout: 5000 });
+          await page.waitForSelector(this.getEnabledCssLocator(locator[i]), { visible: true, timeout: 10000 });
```
Comment (so99y) [inline test/helpers/PlaywrightHelper.js:61]: I am wondering if we could put the value of timeout to config.js

File: playwright-e2e/functional/smoke.spec.ts [hunk 5]
```diff
@@ -2,37 +2,143 @@ import { test, expect } from '../fixtures/fixtures.ts';
...
+    await financialRemedyCourtPage.selectCourtZoneDropDown();
+    await financialRemedyCourtPage.selectHighCourtJudgeLevel(true);
+    await financialRemedyCourtPage.enterSpecialFacilities();
+    await financialRemedyCourtPage.enterSpecialArrangements();
+    await financialRemedyCourtPage.selectShouldNotProceedApplicantHomeCourt(true);
+    await financialRemedyCourtPage.enterFrcReason();
+    await financialRemedyCourtPage.navigateContinue();
+
+    // Has attended miam
+    await miamQuestionPage.selectHasAttenedMiam(true);
```
Comment (ptrelease) [inline playwright-e2e/functional/smoke.spec.ts:109]: Typo here (spelling of attended) if you wanted to correct

**Comment by ptrelease** [comment]
This is great.  Since the team will be using these patterns for future tests, could you change so that the page locators work by type, then name? 

example: https://github.com/hmcts/fpl-ccd-configuration/blob/master/playwright-e2e/pages/base-page.ts

File: playwright-e2e/pages/create-case/PropertyAdjustmentPage.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+    public constructor(page: Page) {
+        super(page);
+
+        this.propertyAdjustmentAddressTextBox = page.getByRole('textbox', {name: 'property address'})
+        this.propertyAdjustmentMortgages = page.getByRole('textbox', {name: 'Name(s) and address(es) of any mortgage(s) for property'})
+        this.propertyAdjustmentAddAdditionalRadio = page.locator(
+          '#additionalPropertyOrderDecision_radio'
+        );
+
```
Comment (ptrelease) [inline playwright-e2e/pages/create-case/PropertyAdjustmentPage.ts:21]: Suggest looking for # locators and changing to type/name ones?

**Review by ptrelease — DISMISSED** [review]

**Review by hamzahtahirhmcts — APPROVED** [review]

**Review by jthmcts — APPROVED** [review]

**Review by ptrelease — APPROVED** [review]

**Pull request merged**